torchgeo/datasets/rioxarray.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""In-memory geographical xarray.DataArray."""
+
+import glob
+import os
+import re
+import sys
+from datetime import datetime
+from typing import Any, Callable, Optional, cast
+
+import netCDF4  # noqa: F401
+import numpy as np
+import torch
+import xarray as xr
+from rasterio.crs import CRS
+from rtree.index import Index, Property
+
+from .geo import GeoDataset
+from .utils import BoundingBox
+
+
+class RioXarrayDataset(GeoDataset):
+    """Wrapper for geographical datasets stored as Xarray Datasets.
+
+    Relies on rioxarray.
```

Comment:
You're currently not using rioxarray, nor do you have a dependency on it

Reply:
The functions in the `__getitem__` method like `ds.rio.crs` or `ds.rio.clip_box` require rioxarray.